src/cirrus/docs/components/tasks/batch.rst

```diff
@@ -416,11 +416,78 @@ CPUs each job requires.
    https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-batch-computeenvironment-computeresources.html
 
 
-Using the AWS spot market
+Using the AWS Spot market
 *************************
 
-.. TODO
-
+Using Spot instances instead of On-Demand can result in a significant cost savings, with only a
+small effect on throughput and latency. Workflow error handling must be well-defined, since Spot
+instances have a significantly higher liklihood of being deallocated as compared with On-Demand
+instances.
+
+This is an example of a ComputeEnvironment that uses Spot::
+
+  FooComputeEnvironment:
+  Type: AWS::Batch::ComputeEnvironment
+  Properties:
+    Type: MANAGED
+    ComputeResources:
+      MinvCpus: ${env:CIRRUS_TASK_FOO_TO_STAC_BATCH_MIN_VCPUS, 0}
+      MaxvCpus: 5000
+      SecurityGroupIds: ${self:custom.batch.SecurityGroupIds}
+      Subnets: ${self:custom.batch.Subnets}
+      InstanceTypes:
+        - r4.xlarge
+        - r4.2xlarge
+        - r4.4xlarge
+        - r4.8xlarge
+        - r4.16xlarge
+        - r5.xlarge
+        - r5.2xlarge
+        - r5.4xlarge
+        - r5.8xlarge
+        - r5.12xlarge
+        - r5.16xlarge
+        - r5.24xlarge
+        - r6i.xlarge
+        - r6i.2xlarge
+        - r6i.4xlarge
+        - r6i.8xlarge
+        - r6i.12xlarge
+        - r6i.16xlarge
+        - r6i.24xlarge
+        - r7i.xlarge
+        - r7i.2xlarge
+        - r7i.4xlarge
+        - r7i.8xlarge
+        - r7i.12xlarge
+        - r7i.16xlarge
+        - r7i.24xlarge
+      Type: SPOT
+      AllocationStrategy: BEST_FIT_PROGRESSIVE
+      SpotIamFleetRole:
+        Fn::GetAtt: [EC2SpotRole, Arn]
+      InstanceRole:
+        Fn::GetAtt: [BatchInstanceProfile, Arn]
+      Tags: { "Name": "Batch Instance - #{AWS::StackName}" }
+      LaunchTemplate:
+        LaunchTemplateId: !Ref FooComputeEnvironmentLaunchTemplate
+        Version: $Latest
+    State: ENABLED
+
+The AllocationStrategy of `BEST_FIT_PROGRESSIVE` indicates that Spot requests should be made
+progressively for the instance types that ECS determines will best meet the resource needs, but
+that any of these machines is acceptable. This is different than the `BEST_FIT` allocation
+strategy that will pick the best one and wait until that best one can be fulfilled.
+
+These specific `InstanceTypes` were chosen because they match the processor architecture needed
+by the Docker image that will run on them (AMD64) and they most closely "pack" with the amount
+vCPU and memory configured for the container, in this case, 1 vCPU and 9.8GB memory, with their
+1 vCPU to 8GB memory ratios.
```

Comment:
If you choose an AllocationStrategy with a wide range of available instance types, larger instances will be running more jobs, and you need to ensure you have appropriate storage for these large instances. For example, if each task requires 3GB of disk space, you may be able to allocate an 8x sized instance to run 8 tasks, but would need 24GB storage. This would be a waste for smaller instance types running 1 task, where the storage is over-allocated.

Reply:
If you choose an AllocationStrategy with a wide range of available instance types,
larger instances will be running more jobs, and you need to ensure you have
appropriate storage for these larger instances. Unfortunately, only one value can be
set for storage, which every EC2 instance gets, no matter how many concurrent tasks it
can support. For example, if each task requires 1vCPU and 3GB of storage and you allow
r4.xlarge (4 vCPUs) to r4.24xlarge (96 vCPUs), you must specify the the maximum amount of
storage required (96 x 3GB = 288GB), even though the r4.xlarge instances will always have
276GB of unused storage that incurs cost.